### PR TITLE
test: Add tests for read_novatel_L1_data_files, and fix broken unittests

### PR DESCRIPTION
**Pull request by madhavmk**
* Add tests for load_novatel() and load_novatel_std()
* Fix broken tests in test_loaders.py
* Add tests for spline_interpolate
* Remove jupyter notebooks Optimized_Kalman.ipynb and kalman_compare.ipynb
* test config.yaml modified to accomodate updated Configuration class
* Fully Addresses Issue #120  and Issue #226, and partially Addresses Issue #240

### CODE REVIEW

File: src/gnatss/posfilter/posfilter.py [hunk 1]
```diff
@@ -64,6 +67,7 @@ def spline_interpolate(
         if col not in constants.PLATFORM_COV_RPH_DIAG:
             result_df[col] = 0.0
 
+    # TODO: Confirm: return values df has different columns depending on full_results
```
Comment (lsetiawan) [inline src/gnatss/posfilter/posfilter.py:70]: Please send me screenshot of it.. I don't quite remember why this is

File: src/gnatss/posfilter/posfilter.py [hunk 2]
```diff
@@ -29,6 +29,8 @@ def spline_interpolate(
         Pandas Dataframe containing Antenna enu directions std deviation Novatel L1 data
     twtt_df : pd.DataFrame
         The travel times data
+    full_result : bool
+        TODO: Add description
```
Comment (lsetiawan) [inline src/gnatss/posfilter/posfilter.py:33]: Add something along the lines of

"Flag to either export the full RPH data from the instrument or just the data that aligns with travel times"

File: src/gnatss/posfilter/posfilter.py [hunk 3]
```diff
@@ -45,6 +47,7 @@ def spline_interpolate(
     # with an 'outer' join, this will result in missing values
     # at points of travel times data that doesn't have corresponding
     # INS RPH data
+    # TODO: Confirm why we are using TT_TIME, GPS_TIME, and RPH_TIME for merge
```
Comment (lsetiawan) [inline src/gnatss/posfilter/posfilter.py:50]: Pleas fix. Thanks!

**Comment by madhavmk** [comment]
assert len(travel_data) == len(spline_interpolate)

**Pull request merged**